### PR TITLE
Inlined.batch import

### PR DESCRIPTION
**Pull request by inlined**
This will help devs write more efficient initial imports. It will also let me add a more efficient implementation for push import since both APNs tokens and subscription APIs have a batch method.

In preparation for the push migration tools I also let migration code return a modified object to be saved back. To prevent this from immediately triggering the beforeSave or migration callbacks I had to add yet another status code =/

### CODE REVIEW

File: cloud/migrator.js [hunk 1]
```diff
@@ -221,6 +225,67 @@ Migrator.prototype.getAfterDelete = function(klass) {
   }
 };
 
+// Get bulk import returns a function that runs either the user's explicit
+// importer or the migration function. Any changes made by the import function
+// plus a change to the object's migration state are applied to the objects,
+// which are saved back in bulk.
+// Returns the number of objects imported.
```
Comment (bklimt) [inline cloud/migrator.js]: More like it returns a function that returns the number of objects imported?

File: cloud/migrator.js [hunk 2]
```diff
@@ -221,6 +225,67 @@ Migrator.prototype.getAfterDelete = function(klass) {
...
+          return migrate(object);
+        }).then(function(maybeChanged) {
+          var ret = maybeChanged instanceof _Parse.Object ? maybeChanged : object;
+          ret.set(consts.MIGRATION_KEY, consts.IMPORTED);
+          return ret;
+        });
+      });
+    } else {
+      migrations = bulkImport(objects).then(function(changed) {
+        return _.map(changed, function(object) {
```
Comment (bklimt) [inline cloud/migrator.js]: You're mapping with a function that implicitly returns undefined. That seems unintentional. Did you mean _.each?

File: cloud/migrator.js [hunk 3]
```diff
@@ -221,6 +225,67 @@ Migrator.prototype.getAfterDelete = function(klass) {
...
+      });
+    }
+
+    return _Promise.when(migrations).then(function(objects) {
+      var saveBack = _Promise.as();
+      for (var start = 0; start < objects.length; start += consts.SAVE_BATCH_SIZE) {
+        var slice = objects.slice(start, start + consts.SAVE_BATCH_SIZE);
+        saveBack = saveBack.then(function() {
+          var p = new _Promise();
+          _Parse.Object.saveAll(slice, {
```
Comment (bklimt) [inline cloud/migrator.js]: It looks to me like saveAll already returns a Promise, at least in https://www.parsecdn.com/js/parse-1.5.0.js

If there's some other reason for this, please add a comment.

**Comment by bklimt** [comment]
LGTM other than the small nits above.

File: cloud/migrator.js (hunk 3 above)
Comment (inlined) [inline cloud/migrator.js]: Nope, just me being forgetful

**Pull request merged**